client/services/code_evaluators/PythonPrereqCheckServiceSpec.js

```diff
@@ -156,4 +156,75 @@ describe('PythonPrereqCheckService', function() {
       expect(codeLibs).toEqual([]);
     });
   });
+
+  describe('getUnsupportedImports', function() {
+    it('correctly returns a list of the unsupported imports', function() {
+      var code = [
+        'import numpy',
+        'import pandas',
+        'import math',
+        '',
+        'def myFunction(arg):',
+        '\treturn arg',
+        ''
+      ].join('\n');
+      var codeLibs = PythonPrereqCheckService.getImportedLibraries(code);
+      expect(PythonPrereqCheckService.getUnsupportedImports(codeLibs)).toEqual([
+        'numpy', 'pandas']);
+    });
+  });
+
+  it('correctly returns an empty list if all imports are supported',
+    function() {
+      var code = [
+        'import math',
+        'import random',
+        'import operator',
+        '',
+        'def myFunction(arg):',
+        '\treturn arg',
+        ''
+      ].join('\n');
+      var codeLibs = PythonPrereqCheckService.getImportedLibraries(code);
+      expect(PythonPrereqCheckService.getUnsupportedImports(codeLibs))
+        .toEqual([]);
+    }
+  );
+
+  describe('checkCode', function() {
+    it(['returns the correct PrereqCheckFailureObject when starter code is ',
+      'missing'].join(''), function() {
+      var starterCode = [
+        'def myFunction(arg):',
+        '\treturn arg',
+        ''
+      ].join('\n');
+      var prereqCheckFailure = PythonPrereqCheckService.checkCode(starterCode,
+          '');
+      expect(prereqCheckFailure.isMissingStarterCode()).toEqual(true);
```

Comment:
Add some additional checks for the case where the starter code has the right method name but the wrong args, and the wrong method name but the right args.